store/src/java/com/zimbra/cs/mailbox/EmailToSMS.java

```diff
@@ -0,0 +1,149 @@
+package com.zimbra.cs.mailbox;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+
+import com.zimbra.cs.account.Account;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.lmtpserver.LmtpCallback;
+import com.zimbra.cs.lmtpserver.ZimbraLmtpBackend;
+import com.zimbra.cs.mime.ParsedMessage;
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.util.ZimbraLog;
+
+public class EmailToSMS implements LmtpCallback {
+
+	private static final EmailToSMS sInstance = new EmailToSMS();
+	private static final String smsDomain = "esms.gov.in";
+
+	private EmailToSMS() {
+	}
+
+	@Override
+	public void afterDelivery(Account account, Mailbox mbox, String envelopeSender, String recipientEmail,
+			Message newMessage) {
+	}
+
+	@Override
+	public void forwardWithoutDelivery(Account account, Mailbox mbox, String envelopeSender, String recipientEmail,
+			ParsedMessage pm) {
+		try {
+			sendEmailSMS(pm, account);
+		} catch (ServiceException e) {
+			ZimbraLog.mailbox.error("Failed to send SMS forwardWithoutDelivery ServiceException  ", e);
+		}
+	}
+
+	private void sendEmailSMS(ParsedMessage pm, Account act) throws ServiceException {
+		String recipients = pm.getRecipients();
+		String[] splitedRecipients = recipients.split(",");
+		String subject = pm.getSubject();
+		String message = pm.getFragment(act.getLocale());
+		String fullMessage = encode(subject + "\n" + message);
+
+		for (int i = 0; i < splitedRecipients.length; i++) {
+			String rcptAddresses = splitedRecipients[i];
+			String[] splitedMobNumber = rcptAddresses.indexOf("<") < 0 ? rcptAddresses.split("@")
+					: rcptAddresses.split("<")[1].replaceAll(">", "").split("@");
+			String mobNumber = splitedMobNumber[0];
+			String mobNumberDomain = splitedMobNumber[1];
+
+			if (mobNumberDomain.toLowerCase().equals(smsDomain)) {
+				call(fullMessage, mobNumber);
+			}
+		}
+	}
+
+	private void call(String message, String mnumber) throws ServiceException {
```

Comment:
we aren't calling the number, just texting it, so better function name would be `text`